### PR TITLE
Don't ignore draw errors

### PR DESCRIPTION
**Pull request by IceSentry**
# Objective

- It's possible to have errors in a draw command, but these errors are ignored

## Solution

- Return a result with the error

## Changelog

Renamed `RenderCommandResult::Failure` to `RenderCommandResult::Skip`
Added a `reason` string parameter to `RenderCommandResult::Failure`

## Migration Guide
If you were using `RenderCommandResult::Failure` to just ignore an error and retry later, use `RenderCommandResult::Skip` instead.

This wasn't intentional, but this PR should also help with https://github.com/bevyengine/bevy/issues/12660 since we can turn a few unwraps into error messages now.

### CODE REVIEW

**Review by jgayfer — APPROVED** [review]
Always nice to have descriptive errors :+1:

**Comment by IceSentry** [comment]
Looks like I should use expect here and just log the errors. I'll update my PR soon.

**Comment by IceSentry** [comment]
Okay, so this generates a bunch of errors now, but it just generates a bunch RenderCommandFailure. So I'll need to also bubble up what those failures are.

**Comment by IceSentry** [comment]
I added a skip variant to `RenderCommandResult` because that's how it's used every where. I also added a new Failure variant with a message so we can log real failures and not just expected situations to skip.

File: crates/bevy_pbr/src/render/mesh.rs [hunk 1]
```diff
@@ -2013,7 +2013,7 @@ impl<P: PhaseItem, const I: usize> RenderCommand<P> for SetMeshBindGroup<I> {
                 It should be set by the prepare_mesh_bind_group system.\n\
                 This is a bevy bug! Please open an issue."
             );
-            return RenderCommandResult::Failure;
+            return RenderCommandResult::Skip;
```
Comment (mockersf) [inline crates/bevy_pbr/src/render/mesh.rs:2016]: should this one be a failure?

**Comment by mockersf** [comment]
> Looks like I should use expect here and just log the errors. I'll update my PR soon.

you kept 3 `expect`, is it... expected?

File: crates/bevy_pbr/src/render/mesh.rs (hunk 1 above)
Comment (IceSentry) [reply crates/bevy_pbr/src/render/mesh.rs:2016]: Probably, I just did a batch rename of every `Failure` to `Skip` to keep the current behaviour

**Comment by IceSentry** [comment]
> you kept 3 expect, is it... expected?

Yes and no, it's possible I missed one, but also, I want this PR to mostly keep the previous behaviour and we can start slowly finding places that should use `Failure` instead of panicking.

File: crates/bevy_pbr/src/render/mesh.rs (hunk 1 above)
Comment (IceSentry) [reply crates/bevy_pbr/src/render/mesh.rs]: done

**Review by janhohenheim — APPROVED** [review]

**Comment by alice-i-cecile** [comment]
@IceSentry let me know once merge conflicts are resolved so I can merge?

**Pull request merged**